kai/reactive_codeplanner/task_manager/task_manager.py

```diff
@@ -319,11 +309,15 @@ def handle_new_tasks_after_processing(self, task: Task) -> None:
                     raise ValueError(
                         f"A task cannot be added as a child of itself: {task}"
                     )
+                chatter.get().chat_simple(f"current child tasks: {len(task.children)}")
```

Comment:
I think this showed me a bug where specific tasks are being added multiple times on retries; I need to figure out whether this is the case. It probably needs to be better printed if when we start working on the depth of a certain task rather then the next high level task IMO.